Sources/Sentry/SentrySession.m

```diff
@@ -55,7 +55,12 @@ - (instancetype)initWithJSONObject:(NSDictionary *)jsonObject
 
         id started = [jsonObject valueForKey:@"started"];
         if ([started isKindOfClass:[NSString class]]) {
-            _started = [NSDate sentry_fromIso8601String:started];
+            NSDate *startedDate = [NSDate sentry_fromIso8601String:started];
+            // When the string can't be converted to a date we don't override the
+            // default value.
+            if (nil != startedDate) {
+                _started = startedDate;
+            }
```

Comment:
M: on Java we'd drop the session though, as `started` is a mandatory field, and if it has an invalid timestamp, `duration` can't be calculated properly as well.

Reply:
I would also like to do that, but we can't return `nil` as this would be a breaking change. See: https://github.com/getsentry/sentry-cocoa/blob/10d7a5b2816201dde26a4fe630fca71dd9f6f06f/Sources/Sentry/SentrySession.m#L41-L46

I put this already down for the next major, see https://github.com/getsentry/sentry-cocoa/discussions/877#discussioncomment-207809

Reply:
I understand its a breaking change but SentrySession is internal right.
`startSession` and `endSession` do not return a Session, nor we do `beforeSend` with Sessions, so I'd prefer to fix the bug as this session would just mess up with session data on the server.
I'm also fine keeping as it is and fixing it on v7 but I'd like to point this out anyway.

Reply:
`SentrySession initWithJSONObject` is public. So making it `nullable` would be a breaking change. We already had this discussion in the past and we agreed with sticking to the current approach and fix it in the next major.

I mean we can argue that this is clearly wrong and we just fix it now. I don't think anyone is using `initWithJSONObject` anyway. Why would they?

Reply:
initializing SentrySession directly isn't really a documented API or expected to be used. I wouldn't bother bumping a major for this breaking change.

Reply:
I agree with Manoel here this will generate trash data and we're better off discarding this instead